domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -268,6 +279,34 @@ constructor(
         useCaseGroup = useCaseGroupBuilder.build()
     }
 
+    @OptIn(ExperimentalCamera2Interop::class)
+    private fun createPreviewUseCase() : Preview {
+        var isPreviewStabilizationSupported = false
+
+        // TODO: find a better way to access camera info of front/back camera before binding
+        //  use cases
+        val availableCameraInfo = cameraProvider.availableCameraInfos
+        if (availableCameraInfo.isNotEmpty()) {
+            for (cameraInfo in availableCameraInfo) {
+                val lensFacing = Camera2CameraInfo.from(cameraInfo).getCameraCharacteristic(
```

Comment:
You can avoid Camera2Interop here by using `CameraSelector.filter`. For instance, you could write something like:

```
val cameraSelector = if (isFrontFacing) {
    CameraSelector.DEFAULT_FRONT_CAMERA
} else {
    CameraSelector.DEFAULT_BACK_CAMERA
}
val isPreviewStabilizationSupported =
    cameraSelector.filter(availableCameraInfo).firstOrNull()?.let {
        Preview.getPreviewCapabilities(it).isStabilizationSupported
    } ?: false
```

Reply:
Thanks. This is much cleaner now to avoid Camera2CameraInfo.